jobrunner/sync.py

```diff
@@ -44,7 +44,9 @@ def sync():
     jobs = find_where(Job, job_request_id__in=job_request_ids)
     jobs_data = [job_to_remote_format(i) for i in jobs]
     log.debug(f"Syncing {len(jobs_data)} jobs back to job-server")
-    api_post("jobs", json=jobs_data)
+
+    if jobs_data:
+        api_post("jobs", json=jobs_data)
```

Comment:
Now I think about it, it might be better to do this check earlier as it would save us a useless query to the db as well.
I was thinking on line 40:
```py
# Bail early if there's nothing to do
if not job_requests:
    return
```

Reply:
done and rebased in 2cb24dc